apps/system/js/updatable.js

```diff
@@ -228,12 +229,69 @@ SystemUpdatable.prototype.handleEvent = function(evt) {
   }
 };
 
-SystemUpdatable.prototype.errorCallBack = function() {
-  UpdateManager.requestErrorBanner();
+SystemUpdatable.prototype.errorCallBack = function(aUpdate) {
+  // Show notification for update installation failures
+  console.debug('Handling systemUpdatable error: updateType',
+                aUpdate.updateType, 'state', aUpdate.state);
+  if (aUpdate.updateType === 'complete' && aUpdate.state === 'failed') {
+    var errorOptions = {
+      bodyL10n: 'systemUpdateErrorDetails',
+      icon: '/style/notifications/images/system_update_error.png',
+      tag: 'systemUpdateError',
+      mozbehavior: {
+        showOnlyOnce: true
+      }
+    };
+
+    NotificationHelper.send('systemUpdateError', errorOptions).then(n => {
+      n.addEventListener('click',
+                         this.showUpdateErrorDetails.bind(this, aUpdate));
+    });
+  } else {
+    // Keep the old banner for the others for now
+    UpdateManager.requestErrorBanner();
+  }
+
   UpdateManager.removeFromDownloadsQueue(this);
   this.downloading = false;
 };
 
+SystemUpdatable.prototype.showUpdateErrorDetails = function(updateError) {
+  var cancel = {
+    title: 'later',
+    callback: function() {
+      Service.request('hideCustomDialog');
+    }
+  };
+
+  var confirm = {
+    title: 'reportNow',
+    callback: function() {
+      Service.request('hideCustomDialog');
+      Notification.get({ tag: 'systemUpdateError' }).then(ns => {
+        ns.forEach(n => {
+          n && n.close();
```

Comment:
Note that using the NotificationHelper will automatically close the notification when you click it, so you don't have to manually do this.

Even so, I would say it's better to just pass the live notification object into showUpdateErrorDetails function (through bind) rather than fetching it again here.